packages/opentelemetry-test-utils/src/instrumentations/instrumentation-singelton.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { InstrumentationBase } from '@opentelemetry/instrumentation';
+
+const OTEL_TESTING_INSTRUMENTATION_SINGLETON = Symbol.for(
+  'opentelemetry.testing.instrumentation_singleton'
+);
+
+type OTelInstrumentationSingeltonGlobal = {
+  [OTEL_TESTING_INSTRUMENTATION_SINGLETON]?: InstrumentationBase;
+};
+const _global = global as OTelInstrumentationSingeltonGlobal;
+
+export const getInstrumentation = <T extends InstrumentationBase>():
+  | T
+  | undefined => {
+  return _global[OTEL_TESTING_INSTRUMENTATION_SINGLETON] as T;
+};
+
+export const registerInstrumentation = <T extends InstrumentationBase>(
```

Comment:
we have `registerInstrumentations` which sounds very similar, to avoid confusion my suggestion would be to just rename to something more meaningful to avoid using this accidentally in real app.

```suggestion
export const registerInstrumentationTesting = <T extends InstrumentationBase>(
```

Reply:
Good point. I choose the name specifically so it matched the name of the function used in the regular, non-testing setup. But now that you mentioned it, I support changing the name to make it explicit that the function is only relevant for tests, and to avoid confusion.